### PR TITLE
code changes to make syncerobject as pointer variable inn fileInode

### PR DESCRIPTION
**Pull request by vadlakondaswetha**
Every fileInode keeps a copy of syncerObject struct instance. To avoid unnecessary memory usage and syncerObject is common to all fileInode, making it a pointer variable.

Note: We cannot keep this instance at fs.go level, since the syncerObject will be different for different buckets incase bucketname is not specified during mount.

Metrics:
Size of single fileInode: ~640 bytes
Copying it by reference decreases the size to ~576 Bytes from ~640 Bytes.
10% reduction in size for 1 fileInode 
For 100K objects, the memory used by fileInodes is 54.93MB which is 26% reduction compared to existing usage 75.05MB

### CODE REVIEW

**Review by raj-prince — COMMENTED** [review]
Change wise looks good. I have some questions:
(1) What happens when we don't pass the bucket name? In that case, what would be the value of syncerBucket?
(2) Inside syncer object we keep appendObjectCreator, which contains a field named prefix, right now it's hard coded. What happens if we start keeping some field specific to object. I haven't find any - but do you know, how we can make sure to not add any gcs.object specific field to syncer/appender object?
(3) Can we add the number in the description, how much memory will be saved with this change?

**Comment by vadlakondaswetha** [comment]
> Change wise looks good. I have some questions: (1) What happens when we don't pass the bucket name? In that case, what would be the value of syncerBucket? (2) Inside syncer object we keep appendObjectCreator, which contains a field named prefix, right now it's hard coded. What happens if we start keeping some field specific to object. I haven't find any - but do you know, how we can make sure to not add any gcs.object specific field to syncer/appender object? (3) Can we add the number in the description, how much memory will be saved with this change?

1. SyncerBucket is an instance of bucket interface. This is also a wrapper over other bucket interface definitions. syncerBucket doesn't store the bucketName variable directly but the underlying wrapper stores it.  BucketName is used to make calls to GCS during http call..
2. We can't store file/object level information in any struct/file other than fileInode. This is by design. With this change, even if they create it will not work as intended since syncerObject instance is common to all fileInodes. It should be discovered during dev testing itself.
3. Done

File: internal/fs/inode/dir.go [hunk 1]
```diff
@@ -531,7 +531,7 @@ func (d *dirInode) readObjects(
 		MaxResults:               MaxResultsForListObjectsCall,
 		// Setting Projection param to noAcl since fetching owner and acls are not
 		// required.
-		ProjectionVal:            gcs.NoAcl,
```
[the remark this answers is not in the document]
Comment (vadlakondaswetha) [reply internal/fs/inode/dir.go:534]: its a formatting change and lint is not complaining. So should be right.

**Review by raj-prince — APPROVED** [review]

**Review by sethiay — APPROVED** [review]
LGTM

**Pull request merged**